cmd/backup/storages/providers.go

```diff
@@ -0,0 +1,87 @@
+package storages
+
+import (
+	"os"
+	"time"
+
+	t "github.com/offen/docker-volume-backup/cmd/backup/types"
+	"github.com/sirupsen/logrus"
+)
+
+// A pool or collection of all implemented storage provider types.
+type StoragePool struct {
```

Comment:
Would it be more flexible if `StoragePool` would just be a slice of things that implement a `StorageBackend` (or similar) interface and then the code just iterates over that list, calling the methods defined on the interface?

Reply:
I thought about that too.
If we only add backends to the pool which are initialized, we can remove the checks `if sp.S3 != nil` inside `CopyAll` and `PruneAll` since it just iterates over the available backends.